runtime/src/bank.rs

```diff
@@ -90,11 +90,17 @@ pub struct Bank {
     last_id_queue: RwLock<LastIdQueue>,
 
     /// Previous checkpoint of this bank
-    parent: Option<Arc<Bank>>,
+    parent: RwLock<Option<Arc<Bank>>>,
 
-    /// Hash of this Bank's state. Only meaningful after freezing it via `new_from_parent()`.
+    /// Hash of this Bank's state. Only meaningful after freezing.
     hash: RwLock<Hash>,
 
+    /// Hash of this Bank's parent's state
+    parent_hash: Hash,
+
+    /// Bank fork id
+    id: u64,
```

Comment:
Why not `slot_height()`?

Reply:
because all that stuff is non-Bank, it's all protocol stuff.  bank just needs an id, and it needs to be something assigned by the callers of new_from_parent().

Reply:
#2884 illustrates this

Reply:
Hmm, I don't agree exactly, but I'll roll with it.